app/routes.js

```diff
@@ -88,20 +88,29 @@ export const routes = (authProvider) => {
 
   router.get('/organisation/:orderId/supplier/search', authProvider.authorise({ claim: 'ordering' }), withCatch(authProvider, async (req, res) => {
     const { orderId } = req.params;
-    const context = await getSupplierSearchPageContext({ orderId });
+    const context = await supplierSearchController.getSupplierSearchPageContext({ orderId });
     res.render('pages/sections/supplier/search/template.njk', addContext({ context, user: req.user, csrfToken: req.csrfToken() }));
   }));
 
   router.post('/organisation/:orderId/supplier/search', authProvider.authorise({ claim: 'ordering' }), withCatch(authProvider, async (req, res) => {
     const { orderId } = req.params;
 
-    const response = validateSupplierSearchForm({ data: req.body });
+    const response = supplierSearchController.validateSupplierSearchForm({ data: req.body });
 
-    const context = await getSupplierSearchPageErrorContext({
-      orderId,
-      validationErrors: response.errors,
+    if (!response.success) {
```

Comment:
I think the pattern we usually use is happy first?

Reply:
yes I noticed this but in this instance this is the validation we are doing. So we don't want to progress any further in the logic of this route if there are validation errors. 

So that's why I've had to change the pattern here slightly.

Reply:
actually going to try this the other way and see how that looks

Reply:
yeah looks better 👍